src/lib/components/WellCompletions/components/Settings/ZoneSelector.test.tsx

```diff
@@ -31,16 +31,39 @@ describe("test Zone Selector", () => {
         });
 
         userEvent.click(screen.getByRole("button", { name: /remove all/i }));
-        const zoneFilter = screen.getByText(/select zone\(s\)\.\.\./i)
-        userEvent.click(zoneFilter)
-        const dropdown = screen.getByPlaceholderText('Search...')
-        await waitFor(() => expect(dropdown).toBeVisible())
-        userEvent.type(dropdown, '{down}')
-        userEvent.type(dropdown, '{enter}')
+        const zoneFilter = screen.getByText(/select zone\(s\)\.\.\./i);
+        userEvent.click(zoneFilter);
+        const dropdown = screen.getByPlaceholderText("Search...");
+        await waitFor(() => expect(dropdown).toBeVisible());
+        userEvent.type(dropdown, "{down}");
+        userEvent.type(dropdown, "{enter}");
 
         expect(testStore.dispatch).toHaveBeenCalledTimes(3);
-        expect(testStore.dispatch).toBeCalledWith({
-            payload: ['zone1'],
+        expect(testStore.dispatch).toHaveBeenNthCalledWith(1, {
+            payload: [],
+            type: "ui/updateFilteredZones",
+        });
+        expect(testStore.dispatch).toHaveBeenNthCalledWith(2, {
+            payload: [],
+            type: "ui/updateFilteredZones",
+        });
```

Comment:
I actually dont understand why the handler get called three times, it looks like it's when the click, {down}, and {enter} actions happen on the SmartNodeSelector. It feels like a small bug to me as it might drag down the performance, might worth asking @rubenthoms.

Reply:
Yes ....It seems like that the keyboard action of down and enter had made the call two times. Thanks for bringing that to notice .